src/app/api/common/delegates/getDelegates.ts

```diff
@@ -135,7 +135,62 @@ async function getDelegates({
   seed?: number;
 }) {
   const pageSize = 20;
-  const { namespace } = Tenant.current();
+  const { namespace, ui } = Tenant.current();
+
+  const allowList = ui.delegates?.allowed || [];
+  const hasAllowList = allowList.length > 0;
+
+  // Applies allow-list filtering to the delegate list
+  const paginatedAllowlistQuery = async (skip: number, take: number) => {
+    switch (sort) {
+      case "most_delegators":
+        return prisma.$queryRawUnsafe<DelegatesGetPayload[]>(
+          `
+            SELECT *
+            FROM ${namespace + ".delegates"}
+            WHERE num_of_delegators IS NOT NULl AND delegate = ANY($1)
+            ORDER BY num_of_delegators DESC
+            OFFSET $2
+            LIMIT $3;
+            `,
+          allowList,
+          skip,
+          take
+        );
+
+      case "weighted_random":
+        await prisma.$executeRawUnsafe(`SELECT setseed($1);`, seed);
+        return prisma.$queryRawUnsafe<DelegatesGetPayload[]>(
+          `
+            SELECT *
+            FROM ${namespace + ".delegates"}
+            WHERE voting_power > 0 AND delegate = ANY($2)
+            ORDER BY -log(random()) / voting_power
+            OFFSET $3
+            LIMIT $4;
+            `,
+          seed,
+          allowList,
+          skip,
+          take
+        );
+
+      default:
+        return prisma.$queryRawUnsafe<DelegatesGetPayload[]>(
+          `
+            SELECT *
+            FROM ${namespace + ".delegates"}
+            WHERE delegate = ANY($1)
+            ORDER BY voting_power DESC
+            OFFSET $2
+            LIMIT $3;
+            `,
+          allowList,
+          skip,
+          take
+        );
```

Comment:
Would it be better to give this default case an explicit label, and error if a sort method wasn't passed?  Forces the caller to be explicit.

Reply:
@jefag can you give me an example?